lib/engine/game/g_18_norway/steps/build_tunnel.rb

```diff
@@ -50,8 +50,8 @@ def build_tunnel(entity, cost)
             entity.spend(cost, @game.bank) if cost.positive?
 
             factory_owner = @game.company_by_id('P3').owner
-            @game.bank.spend(10, factory_owner)
-            @log << "#{factory_owner.name} receives #{@game.format_currency(10)} for building a tunnel"
+            @game.bank.spend(10, factory_owner) unless factory_owner.nil?
+            @log << "#{factory_owner.name} receives #{@game.format_currency(10)} for building a tunnel" unless factory_owner.nil?
```

Comment:
`nil?` calls can be avoided:

```suggestion
            @game.bank.spend(10, factory_owner) if factory_owner
            @log << "#{factory_owner.name} receives #{@game.format_currency(10)} for building a tunnel" if factory_owner
```